src/components/PostDetails.tsx

```diff
@@ -4,97 +4,60 @@ import { NewCommentForm } from './NewCommentForm';
 
 import * as commentsApi from '../api/comments';
 
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import { CommentData } from '../types/Comment';
 
-type Props = {
-  post: Post;
-};
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+
+import {
+  addNewComment,
+  loadComments,
+  removeComment,
+} from '../features/comments/commentsSlice';
+
+export const PostDetails: React.FC = () => {
+  const dispatch = useAppDispatch();
 
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
   const [visible, setVisible] = useState(false);
 
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi
-      .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
-    setVisible(false);
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
-    }
-  };
+  const post = useAppSelector(state => state.selectedPost);
 
-  useEffect(() => {
-    loadComments();
-  }, []);
+  const { comments, loaded, hasError } = useAppSelector(
+    state => state.comments,
+  );
 
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
+  useEffect(() => {
+    if (post?.id) {
+      dispatch(loadComments(post?.id));
+      setVisible(false);
+    }
+  }, [dispatch, post?.id]);
 
   const addComment = async ({ name, email, body }: CommentData) => {
-    try {
-      const newComment = await commentsApi.createComment({
-        name,
-        email,
-        body,
-        postId: post.id,
-      });
-
-      setComments(currentComments => [...currentComments, newComment]);
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
-    } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
+    if (post?.id) {
+      try {
+        const newComment = await commentsApi.createComment({
+          name,
+          email,
+          body,
+          postId: post.id,
+        });
+
+        dispatch(addNewComment(newComment));
+      } catch (error) {}
```

Comment:
Consider adding error handling logic within the catch block to provide user feedback or log the error when adding a new comment fails.